app/models/services/commitment_props.rb

```diff
@@ -57,7 +57,14 @@ def call
                 description: I18n.t('form.commitments.page1.q1.description'),
                 isRequired: true,
                 popupdescription: I18n.t('form.commitments.page1.q1.popupdescription_html'),
-                defaultValue: @commitment.name || ''
+                defaultValue: @commitment.name || '',
+                validators: [
+                  { 
+                    type: 'regex',
+                    text: I18n.t('form.commitments.page1.q1.validation'),  
+                    regex: '^\s*(\S+\s+|\S+$){1,10}$'
```

Comment:
Might be worth extracting either the regex string or the whole validator hash/object into a method at the bottom of the file for each of these regex validators (or one method with an argument of the field name). I like regex, but these look quite complicated at a superficial glance, and it would be good to abstract what they're actually validating for anyone looking at this in the future!